pkg/mutation/mutators/assign/assign_mutator.go

```diff
@@ -57,12 +55,28 @@ func (m *Mutator) TerminalType() parser.NodeType {
 	return schema.Unknown
 }
 
-func (m *Mutator) Mutate(obj *unstructured.Unstructured) (bool, error) {
-	value, err := m.assign.Spec.Parameters.Assign.GetValue(obj)
+func (m *Mutator) Mutate(mutable *types.Mutable) (bool, error) {
+	value, err := m.assign.Spec.Parameters.Assign.GetValue(mutable, m.ID())
 	if err != nil {
 		return false, err
 	}
-	return core.Mutate(m.Path(), m.tester, core.NewDefaultSetter(value), obj)
+	return core.Mutate(m.Path(), m.tester, core.NewDefaultSetter(value), mutable.Object)
+}
+
+func (m *Mutator) ResolveExternalData(data *externaldata.Item) (interface{}, error) {
```

Comment:
If we have a copy of the external data object inside the placeholder, we no longer need to have a `ResolveExternalData` function per-mutator. Calls can be handled inside mutation system, which will make it easier to batch/resolve and reduce code duplication.